SWIG/cashflows.i

```diff
@@ -215,6 +215,32 @@ class FloatingRateCouponPtr : public CouponPtr {
     FloatingRateCouponPtr();
   public:
     %extend {
+        FloatingRateCouponPtr(const Date& paymentDate,
```

Comment:
Is the resulting object useable? It's not an abstract class as I thought, so you can create it, but I guess that any pricer you set will try to downcast the coupon to a more specific type and fail.  Am I wrong?  Do you have a use case for this?

Reply:
In general this whole PR was merely a cleanup on my side. Actually I never really needed the FloatingRateCoupon directly, but mainly for the cast `as_floating_rate_coupon` in the context of Michael von den Drieschs [FlowAnalysis Utility](https://github.com/mdriesch/QuantLib/blob/FlowAnalysis/ql/utilities/flowanalysis.cpp) to debug cashflow objects, if I remember correctly (i.e. iterating over several cast possibilities (`as_coupon`, `as_fixed_rate_coupon`, `as_floating_rate_coupon`) for a given unknown cashflow object and checking the outcome). Apparently this utility tool never made it into the master though.

I fully agree that the object in itself is rather useless due to lack of pricers - it really only provides the encapsulated coupon structure (spread, gearing). If you'd rather not include it, I'd be happy to remove this part of the PR.

Reply:
Yes, I'd rather not have the constructor exported.  I think it would suggest that you can create and use the objects.